### ISSUE TITLE
Wrong plan after the first provisioning of an Open Telemetry plugin

### ISSUE THREAD

**Issue by andrea-migliaccio** [Kong/terraform-provider-konnect#29]
Hi, it's me again :-) 

After provisioning an OpenTelemetry plugin, a subsequent terraform plan shows several changes in the resource even if I haven't made any actual change to the resource definition.

This is just anoying because tf doesn't actually apply anything in reality. So, I'm not blocked by this issue. 

Here's the plan

![image](https://github.com/Kong/terraform-provider-konnect/assets/97603234/1834aaee-1898-4668-9ec8-fc6866c9f066)

Here's the apply after confirming

![image](https://github.com/Kong/terraform-provider-konnect/assets/97603234/4ddc2fba-b724-4239-ab6b-39703610889a)

**Comment by andrea-migliaccio** [comment]
@mheap and this error is popping up again during a tf plan :-( That's even stranger becase it wasn't appening few days ago.

![image](https://github.com/Kong/terraform-provider-konnect/assets/97603234/e1ce3243-4cb8-4fe0-8261-ea6841e1bac9)

**Comment by mheap** [comment]
That is strange! I'll take a look early next week

**Comment by andrea-migliaccio** [comment]
Hi @mheap I may have found what's is causing the issue.

As you may see from the error the provider isn't capable of unmarshalling the "headers" config property. What I've seen is that the OpenTelemetry instance that is causing the issue has

headers: []    // empty array

while others have

headers: null

I've provisioned all of them in the same way but maybe, someone has edited the first one and after that, the null became an empty array. To support this I tried to edit one of the other OT (just saving without changing anything from the edit mask) pluging and I can confirm that the headers property is now and empty array. The same happens to the "Resource Attributes" property.

Now, in order to if I need to manually delete the OT plugins from the portal and from the TF state as well and make TF provision  them again.

**Comment by andrea-migliaccio** [comment]
I can confirm that this procedure (manually delete OT from kong interface and from TF state) has let me proceed, for now.

**Comment by mheap** [comment]
Thank you for the reproduction steps @andrea-migliaccio - I'll get to this in the next week or two

**Comment by mheap** [comment]
I've just reproduced. The schema marks this field as an object with `additionalProperties: true` which means the Go code is doing the correct thing by not deserializing. I'm investigating why the field is being set to `[]` by the UI

**Comment by mheap** [comment]
This is with an internal team to fix in the API itself. Thanks for raising the issue

**Comment by mheap** [comment]
@andrea-migliaccio Good news! I've just tested and this seems to be resolved. Thank you for bringing it to our attention

**Comment by andrea-migliaccio** [comment]
Hi @mheap is this fixing the error due to manual update of the plugin via UI, right? I trust you. I'd need to setup a temporary Control plane to test it :-)

Unfortunatelly within this issue we mixed-up to problems. The "phantom" plans (that I'm still experiencing) are still there (see the original issue post).

**Comment by mheap** [comment]
Yep, the error due to manual update. Let me investigate the phantom plan

**Comment by mheap** [comment]
@andrea-migliaccio Could you share your `konnect_gateway_plugin_opentelemetry` resource config? I can't reproduce with a minimal config:

```hcl
resource "konnect_gateway_plugin_opentelemetry" "my_opentelemetry" {
  enabled = true

  config = {
    endpoint = "http://example.com"
  }

  control_plane_id = konnect_gateway_control_plane.tfdemo.id
  service = {
    id = konnect_gateway_service.httpbin.id
  }
}
```

**Comment by mheap** [comment]
Please let us know if this is still an issue

**Comment by andrea-migliaccio** [comment]
Hi @mheap sorry, I was in vacation in the last weeks :-)

`resource "konnect_gateway_plugin_opentelemetry" "opentelemetry_plugin" {
  config = {
    batch_flush_delay =  null
    batch_span_count =  null
    connect_timeout =  1000
    endpoint =  "http://datadog-otel-collector-endpoint:4318/v1/traces"
    header_type = "preserve"
    http_response_header_for_traceid =  null
    propagation =  {
      clear =  null
      default_format =  "b3"
      extract =  ["b3","datadog"]
      inject = ["b3","datadog"]
    }
    queue =  {
      initial_retry_delay =  0.01
      max_batch_size =  1
      max_bytes =  null
      max_coalescing_delay =  1
      max_entries =  10000
      max_retry_delay =  60
      max_retry_time =  60
    }
    read_timeout =  5000
    sampling_rate =  1
    send_timeout =  5000
  }

  lifecycle {
    ignore_changes = [
      config[0].headers,
      config[0].resource_attributes
    ]
  }

  enabled = true
  protocols =  ["grpc","grpcs","http","https"]

  // Enable the resource only if var.control-plane-name contains the string "global-plugins"
  count = var.control-plane-global == true ? 1 : 0
  control_plane_id = konnect_gateway_control_plane.kcp.id
}`

And here the resulting terraform plan of one of the resources. We have 6 plugins provisioned by this same definition using the count/array facility of terraform (as it can be noticed by the very last two lines of the snippet).

![image](https://github.com/user-attachments/assets/3893c036-ab94-437c-ba12-19e695f7d802)

**Comment by andrea-migliaccio** [comment]
Hi @mheap , I don't know why but I've just noticed that this is not happening anymore. On the other side it's happening with the FileLog Plugin :-) I'm opening a dedicated issue
